lib/helpers/cursor/eachAsync.js

```diff
@@ -23,22 +23,20 @@ const promiseOrCallback = require('../promiseOrCallback');
 
 module.exports = function eachAsync(next, fn, options, callback) {
   const parallel = options.parallel || 1;
-  const batchSize = options.batchSize;
+  const batchSize = options.batchSize || undefined;
   const enqueue = asyncQueue();
 
-  return promiseOrCallback(callback, cb => {
-    if (batchSize != null) {
-      if (typeof batchSize !== 'number') {
-        throw new TypeError('batchSize must be a number');
-      }
-      if (batchSize < 1) {
-        throw new TypeError('batchSize must be at least 1');
-      }
-      if (batchSize !== Math.floor(batchSize)) {
-        throw new TypeError('batchSize must be a positive integer');
-      }
+  if (typeof batchSize !== 'undefined') {
```

Comment:
Why move this out of `promiseOrCallback()`? We prefer to consistently return a promise, rather than sometimes return promise vs sometimes throwing a sync error - more like `async function()` semantics

Reply:
Oh nice. Didn't thought about this.

Reply:
fixed